module/Frontpage/src/Frontpage/Service/Poll.php

```diff
@@ -25,10 +25,19 @@ public function getNewestPoll()
      * Retrieves a poll by its id
      * @param int $pollId the id of the poll to retrieve
      * @return PollModel|null
+     *
+     * @throws \User\Permissions\NotAllowedException if the user isn't allowed to see unapproved polls
      */
     public function getPoll($pollId)
     {
-        return $this->getPollMapper()->findPollById($pollId);
+        $poll = $this->getPollMapper()->findPollById($pollId);
+        if ((null === $poll->getApprover()) && !$this->isAllowed('view_unapproved')) {
```

Comment:
In the same file is_null is used consequently, that is what I meant. Personally I don't mind if $a === null and is_null($a) is used interchangeably, since both can be understood easily (Although a lot of people would mind). Yoda conditions I do mind.